spring-cloud-gcp-security-iap/src/main/java/org/springframework/cloud/gcp/security/iap/AudienceValidator.java

```diff
@@ -55,16 +57,17 @@ public AudienceValidator(AudienceProvider audienceProvider) {
 
 	@Override
 	public OAuth2TokenValidatorResult validate(Jwt t) {
-		if (t.getAudience() != null && t.getAudience().contains(this.audience)) {
-			return OAuth2TokenValidatorResult.success();
-		}
-		else {
-			if (LOGGER.isWarnEnabled()) {
-				LOGGER.warn(String.format(
-						"Expected audience %s did not match token audience %s", this.audience, t.getAudience()));
+		String[] audiences = StringUtils.trimArrayElements(StringUtils.commaDelimitedListToStringArray(this.audience));
+		for (String audience : audiences) {
+			if (t.getAudience() != null && t.getAudience().contains(audience)) {
```

Comment:
Now that this is done in a loop, it makes sense to move `t.getAudience() != null` check to the very beginning of `validate()` method and return `OAuth2TokenValidatorResult.failure(INVALID_AUDIENCE);` right away if the audience is missing in token.